templates/nro/.circleci/config.yml.tmpl

```diff
@@ -23,6 +23,8 @@ job_environments:
     WP_DB_NAME: {{ .Env.MYSQL_USERNAME }}_{{ .Env.MYSQL_DATABASE }}_develop
     WP_STATELESS_BUCKET: {{ .Env.CONTAINER_PREFIX }}-stateless-develop
 {{- if isTrue .Env.MAKE_RELEASE }}
+  release_build_env: &release_build_env
+    GOOGLE_PROJECT_ID: {{ .Env.GCP_PRODUCTION_PROJECT }}
```

Comment:
I kept `GOOGLE_PROJECT_ID` in the release and production environments, instead of removing it (or other common ones). That way we don't have to add this new set of vars to all those jobs in this PR, which would add quite a lot of changes to review, and reviewing these templatized config files is already challenging enough :smile: